debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java

```diff
@@ -394,7 +400,34 @@ protected void complete(SnapshotContext<MySqlOffsetContext> snapshotContext) {
      */
     @Override
     protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<MySqlOffsetContext> snapshotContext, TableId tableId) {
-        return Optional.of(String.format("SELECT * FROM `%s`.`%s`", tableId.catalog(), tableId.table()));
+        String snapshotSelectColumns = getSnapshotSelectColumns(tableId);
+        return Optional.of(String.format("SELECT %s FROM `%s`.`%s`", snapshotSelectColumns, tableId.catalog(), tableId.table()));
+    }
+
+    @Override
+    protected String getSnapshotSelectColumns(TableId tableId) {
+        Table table = databaseSchema.tableFor(tableId);
+        List<String> columnNames = table.retrieveColumnNames().stream()
+                .filter(columnName -> connectorConfig.getColumnFilter().matches(tableId.catalog(), tableId.schema(), tableId.table(), columnName))
+                .collect(Collectors.toList());
+
+        if (columnNames.isEmpty()) {
+            LOGGER.info("All columns in table {} were excluded due to include/exclude lists, defaulting to selecting primary keys only", tableId);
+            columnNames = table.primaryKeyColumnNames();
+        }
```

Comment:
All this logic (filtering, falling back to PK columns) is pretty much the same across connectors, no? So can we somehow extract that? Perhaps the right contract for this method would be something like this:

```
List<String> prepareSnapshotSelectColumns(List<String> columns)
```

I.e. only leaving the actual column name quoting or whatever else is needed to implmentations? Or perhaps this even:

```
@FunctionalInterface
interface ColumnNamePreparer {
    String getPreparedColumnName(String columnName);
}

And then this in the connector-specic sources:

```
ColumnNamePreparer getColumnNamePreparer();
```

WDYT?